### ISSUE TITLE
Dependency on mscorlib

### ISSUE THREAD

**Issue by brentarias** [reactjs/React.NET#282]
I've created a brand-new ASP.NET Core RC2 project in Visual Studio 2015.  When I try to add the React.NET middleware, I'm given an error that indicates mscorlib is required.

![image](https://cloud.githubusercontent.com/assets/1289327/16096631/b2a1eb38-32fe-11e6-909c-db8b7c86a6b7.png)

Since [ReactJS.Net compatibility with ASP.NET Core RC2 was officially announced](http://reactjs.net/2016/05/2.4.0-release.html), I would expect there to be no dependencies on mscorlib. 

This is perhaps tangentially related to #275, which for me was resolved by changing my project.json to include "net46":

    "frameworks": {
      "netcoreapp1.0": {
        "imports": [
          "net46",
          "dotnet5.6",
          "dnxcore50",
          "portable-net45+win8"
        ]
      }
    },

Unfortunately the current error, which states I must include mscorlib, is unsolved.  I don't know how to "add mscorlib".  The intellisense action will not perform the fix for me, and I can't manually add mscorlib to my project.json...evidently there is no such package (historically it has been merely an assembly kept in the GAC).

In short, it looks as though ReactJS.Net simply does not at all work with ASP.NET Core RC2.

**Comment by shahid-pk** [comment]
`"frameworks": {
  "net45": {
       ReactJS.NET : "2.4.0 ",
  }
},`

change to this, this should work. you should be targeting .net 4.5 instead of .net core.

**Comment by brentarias** [comment]
From what I understand, if React.AspNet were rebuilt to target netstandard 1.1 then it can be used in .net 4.5 and in .net core both.  Are there any timelines or existing feature requests for achieving that?

**Comment by Daniel15** [comment]
Duplicate of #275. This is expected at the moment. There's currently no JavaScript engine that runs on .NET Core. You will need to switch to using the full .NET Framework ("net451" rather than "netcoreapp1.0" in your project.json). Hopefully someone can get Clearscript running on .NET Core at some point. 

I remember seeing a comment where someone mentioned they got VroomJs running on .NET Core. I haven't had time to test that out myself though, but that is definitely one option :)

**Comment by brentarias** [comment]
Then the real issue is, [why does the main ReactJS.Net site declare that ReactJS.Net is now ready for RC2?](http://reactjs.net/2016/05/2.4.0-release.html).  The official site says it has been [available for RC2 since May 24th](http://reactjs.net/).

Given that Clearscript is not yet ready for RC2 (and thus affecting ReactJS.Net), does it make sense to leverage [edge.js](https://github.com/tjanczuk/edge) instead?  Edge.js will be [available for RC2 by June 22nd](https://github.com/tjanczuk/edge/pull/442).

**Comment by Daniel15** [comment]
>  why does the main ReactJS.Net site declare that ReactJS.Net is now ready for RC2?. The official site says it has been available for RC2 since May 24th.

You're conflating ASP.NET Core and .NET Core :) ReactJS.NET works fine with ASP.NET Core, but you must use the full .NET Framework (`net451` in your `project.json). It does not work with .NET Core at this point in time.

**Comment by brentarias** [comment]
???

I'm trying to use ReactJS.NET with **ASP.NET Core** right now...and I'm using the configuration suggested by shahid-pk...and *it isn't working*.  That's why this whole thread is here in the first place.

Furthermore, as of RC2...both .NET Core and ASP.NET Core have been merged into one runtime.  If ReactJS.Net is working with ASP.NET Core RC2, then *ipso facto*, it is already working with .NET Core.  Said again: as of RC2, it is no longer possible to make an assembly for ASP.NET Core that is incompatible with .NET Core.  But even if you think that statement is incorrect...

To circle back...ReactJS.Net isn't working with **ASP.NET** Core at all.  Again, that was the whole point of this issue.

**Comment by Daniel15** [comment]
> as of RC2, it is no longer possible to make an assembly for ASP.NET Core that is incompatible with .NET Core

That's not true :)

A project with `net451` in its project.json will **only** execute on .NET Framework, not on .NET Core. See the table at https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md, and the issue at https://github.com/aspnet/Hosting/issues/745 where someone asked a similar question.

To make a website that runs on .NET Framework rather than .NET Core, you need to add `net451

> both .NET Core and ASP.NET Core have been merged into one runtime

.NET Core is the actual runtime (including the garbage collector and core libraries), whereas ASP.NET Core is built on top of that. ASP.NET Core runs on top of both .NET Core **and** .NET Framework 4.5.

**Comment by Daniel15** [comment]
> as of RC2, it is no longer possible to make an assembly for ASP.NET Core that is incompatible with .NET Core

That's not true :)

A project with `net451` in its project.json will **only** execute on .NET Framework, not on .NET Core. See the table at https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md, and the issue at https://github.com/aspnet/Hosting/issues/745 where someone asked a similar question.

To make a website that runs on .NET Framework rather than .NET Core, you need to add `net451` to its `framework` section in `project.json`. Here's an example site (my personal site/blog) that uses ASP.NET RC2 on .NET Framework: https://github.com/Daniel15/Website/blob/master/Daniel15.Web/project.json. The sample site bundled with ReactJS.NET also uses .NET Framework rather than .NET Core: https://github.com/reactjs/React.NET/blob/master/src/React.Sample.Mvc6/project.json

> both .NET Core and ASP.NET Core have been merged into one runtime

.NET Core is the actual runtime (including the garbage collector and core libraries), whereas ASP.NET Core is built on top of that. ASP.NET Core runs on top of both .NET Core **and** .NET Framework 4.5.

**Comment by brentarias** [comment]
First, thank you for your patience.  Second, Man this is confusing as hell.  I admit I've consulted the [net-platform-standard page you referenced](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md) numerous times over the last three months, and it has never made any sense to me.

Bottom line: do you have any idea of why intellisense is telling me that my project.json is full of illegal dependencies when I select framework `net451`?
